sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -916,6 +919,9 @@ def _use_unified_worker(pipeline_options):
   debug_options = pipeline_options.view_as(DebugOptions)
   use_unified_worker_flag = 'use_unified_worker'
 
+  if debug_options.lookup_experiment(use_unified_worker_flag):
```

Comment:
PTAL at this method - we need to remove some of the statements in 922-928.

Reply:
Do we need 922-924?

Reply:
nvm, reading this again, looks like this logic is intenional

Reply:
Ok, I think I understand the intent but I think it is confusing - can we move lines 925-926 to dataflow_runner.py, see: https://github.com/apache/beam/blob/1117508136ac5d0d66141c0e8f1c0830fb223a3a/sdks/python/apache_beam/runners/dataflow/dataflow_runner.py#L480, and replace  920-928 with line 928?

Reply:
We can move 925-926 to dataflow_runner.py but then i feel the logic to determine unified_worker usage will be more staggered. 
I think it will be better to keep the interpretation of `use_unified_worker` and `use_runner_v2` at a single place.
For now, we support both flag but I think we will transition to `use_runner_v2` over time (not sure by when).

Reply:
I think that it would be cleaner to not modify experiments in `_use_unified_worker`, since the name of this method does not suggest that it modifies the state, and it would also be easier to follow the logic. Also, if Dataflow runner relies on the fact that "use_runner_v2" experiment is set, it might be better to set it explicitly at some point, instead of relying on the fact that "_use_unified_worker" will be called. Leaving this up to you.